packages/flutter/lib/src/rendering/custom_layout.dart

```diff
@@ -120,37 +119,36 @@ abstract class MultiChildLayoutDelegate {
   /// Creates a layout delegate.
   ///
   /// The layout will update whenever [relayout] notifies its listeners.
-  MultiChildLayoutDelegate({ Listenable relayout }) : _relayout = relayout;
+  MultiChildLayoutDelegate({ Listenable? relayout }) : _relayout = relayout;
 
-  final Listenable _relayout;
+  final Listenable? _relayout;
 
-  // TODO(ianh): make these late final
-  /*late*/ Map<Object/*!*/, RenderBox>/*!*/ _idToChild;
-  /*late*/ Set<RenderBox/*!*/>/*!*/ _debugChildrenNeedingLayout;
+  Map<Object, RenderBox>? _idToChild;
+  Set<RenderBox>? _debugChildrenNeedingLayout;
 
   /// True if a non-null LayoutChild was provided for the specified id.
   ///
   /// Call this from the [performLayout] or [getSize] methods to
   /// determine which children are available, if the child list might
   /// vary.
-  bool hasChild(Object childId) => _idToChild[childId] != null;
+  bool hasChild(Object childId) => _idToChild![childId] != null;
 
   /// Ask the child to update its layout within the limits specified by
   /// the constraints parameter. The child's size is returned.
   ///
   /// Call this from your [performLayout] function to lay out each
   /// child. Every child must be laid out using this function exactly
   /// once each time the [performLayout] function is called.
-  Size/*!*/ layoutChild(Object childId, BoxConstraints constraints) {
-    final RenderBox child = _idToChild[childId];
+  Size layoutChild(Object childId, BoxConstraints constraints) {
+    final RenderBox? child = _idToChild![childId];
     assert(() {
       if (child == null) {
         throw FlutterError(
           'The $this custom multichild layout delegate tried to lay out a non-existent child.\n'
           'There is no child with the id "$childId".'
         );
       }
-      if (!_debugChildrenNeedingLayout.remove(child)) {
+      if (!_debugChildrenNeedingLayout!.remove(child)) {
```

Comment:
_debugChildrenNeedingLayout should be non-nullable

Reply:
It can't because it may be used before its first initialization.

```
    Set<RenderBox>? debugPreviousChildrenNeedingLayout;
    assert(() {
      debugPreviousChildrenNeedingLayout = _debugChildrenNeedingLayout;
      _debugChildrenNeedingLayout = <RenderBox>{};
      return true;
    }());
```

Reply:
k